bug: nil pointer dereference when two deferred interface method calls exist in the same function

When a function has two defer statements that call methods on different interface values, a nil pointer dereference occurs during the deferred execution (after the function body completes). Wrapping either deferred call in a closure eliminates the panic.

Minimal reproducer:
```go
package main

import "fmt"

type Closer interface {
    Close() error
}

type A struct{ name string }
func (a *A) Close() error { fmt.Println("closing", a.name); return nil }

type B struct{ name string }
func (b *B) Close() error { fmt.Println("closing", b.name); return nil }

func main() {
    var x Closer = &A{name: "first"}
    defer x.Close()

    var y Closer = &B{name: "second"}
    defer y.Close()

    fmt.Println("body done")
}
```

```shell
$ tinygo run main.go
body done
closing second
panic: nil pointer dereference
```

Expected behavior: Both deferred calls execute without error, identical to `go run main.go`:

```shell
body done
closing second
closing first
```

Workaround: Wrapping either defer in a closure avoids the bug:

```go
defer func() { x.Close() }()
```

Notes:

- The panic occurs only when both deferred calls are direct interface method calls. If one or both are wrapped in closures, it works correctly.
- It does not matter what the interface is or whether the two values share the same concrete type — the panic reproduces as long as two distinct interface method defers coexist.
- Same concrete type (two *A values) also triggers it.
- The issue is not GC-related (-gc=leaking still panics).

A better reproducer:

```
~/go/src/github.com/dgryski/bug/wazy $ cat main.go
package main

import "context"

type Closer interface {
	Close(ctx context.Context) error
}

type Runtime interface {
	Foo()
	Closer
}

type runtime struct{}

func (r *runtime) Foo()                            {}
func (r *runtime) Close(ctx context.Context) error { return nil }

type module struct{}

func (m *module) Close(ctx context.Context) error { return nil }

func run() {
	ctx := context.Background()
	var r Runtime = &runtime{}
	defer r.Close(ctx)

	var m Closer = &module{}
	defer m.Close(ctx)
}

func main() {
	run()
	println("done")
}
~/go/src/github.com/dgryski/bug/wazy $ tinygo run main.go
panic: nil pointer dereference
failed to run compiled binary /var/folders/yh/8gzm4zjj0zzdm60t2mm9mqkc0000gn/T/tinygo4231916622/main: signal: abort trap
```

--- a/compiler/defer.go
+++ b/compiler/defer.go
@@ -372,12 +372,12 @@ func (b *builder) createDefer(instr *ssa.Defer) {
 		// Method call on an interface.
 
 		// Get callback type number.
-		methodName := instr.Call.Method.FullName()
-		if _, ok := b.deferInvokeFuncs[methodName]; !ok {
-			b.deferInvokeFuncs[methodName] = len(b.allDeferFuncs)
+		key := b.getInvokeFunctionName(&instr.Call)
+		if _, ok := b.deferInvokeFuncs[key]; !ok {
+			b.deferInvokeFuncs[key] = len(b.allDeferFuncs)
 			b.allDeferFuncs = append(b.allDeferFuncs, &instr.Call)
 		}
-		callback := llvm.ConstInt(b.uintptrType, uint64(b.deferInvokeFuncs[methodName]), false)
+		callback := llvm.ConstInt(b.uintptrType, uint64(b.deferInvokeFuncs[key]), false)
 
 		// Collect all values to be put in the struct (starting with
 		// runtime._defer fields, followed by the call parameters).

--- a/compiler/interface.go
+++ b/compiler/interface.go
@@ -1193,8 +1193,7 @@ func (c *compilerContext) getMethodSetValue(methods []*types.Func) llvm.Value {
 // thunk is declared, not defined: it will be defined by the interface lowering
 // pass.
 func (c *compilerContext) getInvokeFunction(instr *ssa.CallCommon) llvm.Value {
-	s, _ := c.getTypeCodeName(instr.Value.Type().Underlying())
-	fnName := s + "." + instr.Method.Name() + "$invoke"
+	fnName := c.getInvokeFunctionName(instr)
 	llvmFn := c.mod.NamedFunction(fnName)
 	if llvmFn.IsNil() {
 		sig := instr.Method.Type().(*types.Signature)
@@ -1211,6 +1210,11 @@ func (c *compilerContext) getInvokeFunction(instr *ssa.CallCommon) llvm.Value {
 		llvmFn.AddFunctionAttr(c.ctx.CreateStringAttribute("tinygo-methods", methods))
 	}
 	return llvmFn
+}
+
+func (c *compilerContext) getInvokeFunctionName(instr *ssa.CallCommon) string {
+	s, _ := c.getTypeCodeName(instr.Value.Type().Underlying())
+	return s + "." + instr.Method.Name() + "$invoke"
 }
 
 // createInterfaceTypeAssert creates a call to a declared-but-not-defined

--- a/testdata/calls.go
+++ b/testdata/calls.go
@@ -70,6 +70,7 @@ func main() {
 
 	// regression testing
 	regression1033()
+	regression5541()
 
 	//Test deferred builtins
 	testDeferBuiltinClose()
@@ -222,6 +223,46 @@ func foo(bar *Bar) error {
 	defer c.Close()
 
 	return nil
+}
+
+var regression5541RuntimeClosed, regression5541ModuleClosed int
+
+type regression5541Closer interface {
+	Close()
+}
+
+type regression5541Runtime interface {
+	Foo()
+	regression5541Closer
+}
+
+type regression5541RuntimeImpl struct{}
+
+func (*regression5541RuntimeImpl) Foo() {
+}
+
+func (*regression5541RuntimeImpl) Close() {
+	regression5541RuntimeClosed++
+}
+
+type regression5541Module struct{}
+
+func (*regression5541Module) Close() {
+	regression5541ModuleClosed++
+}
+
+func regression5541() {
+	func() {
+		var runtime regression5541Runtime = &regression5541RuntimeImpl{}
+		defer runtime.Close()
+
+		var module regression5541Closer = &regression5541Module{}
+		defer module.Close()
+	}()
+
+	if regression5541RuntimeClosed != 1 || regression5541ModuleClosed != 1 {
+		println("deferred interface methods were not both called")
+	}
 }
 
 type issue1304 struct {

